code/android-core-library/src/main/java/com/adobe/marketing/mobile/util/TimeUtils.kt

```diff
@@ -36,45 +39,69 @@ object TimeUtils {
     }
 
     /**
-     * Gets the the ISO 8601 formatted date `String` for the current date.
-     * TimeZone format used is RFC822 using ZZZ formatting letter. ex: 9th July 2020 PDT will be formatted as 2020-07-09T15:09:18-0700.
+     * Gets the ISO 8601 formatted with RFC 822 time zone, second precision date `String` for the provided
+     * date using the current device time zone.
+     * Date pattern used is [DatePattern.ISO8601_TIMEZONE_RFC822_PRECISION_SECOND]
+     * which has timezone RFC 822 'Z' pattern, which gives a timezone of ex: "-0700" ("+0000" for UTC +0)
+     * Ex (device in time zone "America/Los_Angeles"): Wed Nov 30 11:53:09.497 GMT-07:00 2022 -> 2022-11-30T11:53:09-0700
      *
-     * @return Iso8601 formatted date [String]
+     * @param date the [Date] to apply the formatting to; defaults to the current [Date]
+     * @return date [String] formatted as ISO 8601 with RFC 822 time zone, second precision
      */
     @JvmStatic
-    fun getIso8601Date(): String? {
-        return getIso8601Date(Date(), ISO8601_DATE_FORMATTER_TIMEZONE_RFC822)
+    @JvmOverloads
+    fun getIso8601DateTimeZoneRFC822(date: Date? = Date()): String? {
```

Comment:
Can you make data non optional?

Reply:
Can you change the implementation to 
```
fun getIso8601DateTimeZoneRFC822(date: Date = Date()): String {
	return getFormattedDate(date,DatePattern.ISO8601_TIMEZONE_RFC822_PRECISION_SECOND) ?? ""
}
```
Also make similar changes to other functions